src/ostree/ot-builtin-summary.c

```diff
@@ -107,6 +107,80 @@ ostree_builtin_summary (int argc, char **argv, GCancellable *cancellable, GError
             return FALSE;
         }
 
+#ifdef OSTREE_ENABLE_EXPERIMENTAL_API
+      const char *collection_id = ostree_repo_get_collection_id (repo);
+#else  /* if !OSTREE_ENABLE_EXPERIMENTAL_API */
+      const char *collection_id = NULL;
+#endif  /* OSTREE_ENABLE_EXPERIMENTAL_API */
+
+      /* Write out a new metadata commit for the repository. */
+      if (collection_id != NULL)
+        {
+#ifdef OSTREE_ENABLE_EXPERIMENTAL_API
+          OstreeCollectionRef collection_ref = { (gchar *) collection_id, (gchar *) OSTREE_REPO_METADATA_REF };
+          g_autofree char *old_ostree_metadata_checksum = NULL;
+          g_autofree gchar *new_ostree_metadata_checksum = NULL;
+          g_autoptr(OstreeMutableTree) mtree = NULL;
+          g_autoptr(OstreeRepoFile) repo_file = NULL;
+          g_autoptr(GVariantDict) new_summary_commit_dict = NULL;
+          g_autoptr(GVariant) new_summary_commit = NULL;
+
+          if (!ostree_repo_resolve_rev (repo, OSTREE_REPO_METADATA_REF,
+                                        TRUE, &old_ostree_metadata_checksum, error))
+            return FALSE;
+
+          /* Add bindings to the metadata. */
+          new_summary_commit_dict = g_variant_dict_new (additional_metadata);
+          g_variant_dict_insert (new_summary_commit_dict, OSTREE_COMMIT_META_KEY_COLLECTION_BINDING,
+                                 "s", collection_ref.collection_id);
+          g_variant_dict_insert_value (new_summary_commit_dict, OSTREE_COMMIT_META_KEY_REF_BINDING,
+                                       g_variant_new_strv ((const gchar * const *) &collection_ref.ref_name, 1));
+          new_summary_commit = g_variant_dict_end (new_summary_commit_dict);
+
+          if (!ostree_repo_prepare_transaction (repo, NULL, cancellable, error))
+            return FALSE;
+
+          mtree = ostree_mutable_tree_new ();
+          if (!ostree_repo_write_dfd_to_mtree (repo, AT_FDCWD, ".", mtree, NULL, NULL, error))
```

Comment:
Hmm?  I don't think we should be relying on the process' cwd here, right?  It looks like that cwd is the entire unit test directory at least from the tests.

Are you intending to include *any* content in the metadata ref?  If not, the libarchive code has some "create empty directory" code.

Reply:
Oops. The intention is to not include any content in the commit, correct. I can’t see anything immediately obvious for this in the libarchive code. Which code do you mean, exactly?

I need to fix flatpak to not use `AT_FDCWD, "."` too, once this is done.

Reply:
It's this bit:

```
  /* If we didn't import anything at all, and autocreation of parents
   * is enabled, automatically create a root directory.  This is
   * useful primarily when importing Docker image layers, which can
   * just be metadata.
   */
  if (opts->autocreate_parents &&
      ostree_mutable_tree_get_metadata_checksum (mtree) == NULL)
    {
      glnx_unref_object GFileInfo *fi = g_file_info_new ();
      g_file_info_set_attribute_uint32 (fi, "unix::uid", 0);
      g_file_info_set_attribute_uint32 (fi, "unix::gid", 0);
      g_file_info_set_attribute_uint32 (fi, "unix::mode", DEFAULT_DIRMODE);

      g_autoptr(GFileInfo) mfi = NULL;
      (void)_ostree_repo_commit_modifier_apply (self, modifier, "/",
                                                fi, &mfi);

      if (!aic_ensure_parent_dir_with_file_info (&aictx, mtree, "/", mfi, NULL,
                                                 cancellable, error))
        goto out;
    }

```

Reply:
Great, thanks. I’ve updated the patch and added a test.

Reply:
I’ll fix flatpak tomorrow; need to head off now.